mathics/core/parser/parser.py

```diff
@@ -308,13 +308,33 @@ def p_RawLeftAssociation(self, token):
     def p_LeftRowBox(self, token):
         self.consume()
         children = []
-        self.box_depth += 1
-        self.bracket_depth += 1
+        # If this does not happend, it would be because
+        # it was called when a `FormBox` was found.
+        if token.tag == "LeftRowBox":
+            self.box_depth += 1
+            self.bracket_depth += 1
+
         token = self.next()
-        while token.tag not in ("RightRowBox", "OtherscriptBox"):
+        while token.tag not in ("RightRowBox", "OtherscriptBox", "FormBox"):
             newnode = self.parse_box(0)
             children.append(newnode)
             token = self.next()
+
+        if token.tag == "FormBox":
+            if len(children) == 0:
+                fmt = Symbol("StandardForm")
+            elif len(children) == 1:
+                fmt = children[0]
+                if type(fmt) is String:
+                    fmt_name = fmt.value
+                    if is_symbol_name(fmt_name):
+                        fmt = Symbol(fmt_name)
+                    else:
+                        fmt = Node("Removed", Symbol("$$Failure"))
```

Comment:
@rocky,you are right: it should be 
```
fmt = Node("Removed", String("$$Failure"))
```
to be compatible with WMA.

For example:

```
In WMA, if `FormBox` receives wrong arguments (a Number as a format for instance) the interpreter produces
```
In[1]:= \(3 \` + b\)                                                            

Out[1]= FormBox[RowBox[{+, b}], Removed[$$Failure]]

In[2]:= %//FullForm                                                             

Out[2]//FullForm= FormBox[RowBox[List["+", "b"]], Removed["$$Failure"]]
```

Reply:
Fixed in [f6f3d1f](https://github.com/Mathics3/mathics-core/pull/545/commits/f6f3d1fc29ed338d92829377f77d6a521e099129)